plugins/in_systemd/systemd.c

```diff
@@ -276,15 +282,28 @@ static int in_systemd_collect(struct flb_input_instance *i_ins,
     if (ret_j == 0) {
         ctx->pending_records = FLB_FALSE;
         return FLB_SYSTEMD_OK;
-    }
-
+    } else if (ret_j > 0) {
     /*
-     * ret_j == -1, the loop was broken due to some special condition like
+     * ret_j == 1, but the loop was broken due to some special condition like
      * buffer size limit or it reach the max number of rows that it supposed to
      * process on this call. Assume there are pending records.
      */
     ctx->pending_records = FLB_TRUE;
     return FLB_SYSTEMD_MORE;
+    } else {
```

Comment:
Please add the "else" statement in a new line, according to:

https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md